src/components/UsedCoupons/index.jsx

```diff
@@ -0,0 +1,216 @@
+import React, { useState, useEffect, useRef } from "react";
+import "./style.css";
+import {
+  Table,
+  Input,
+  Button,
+  Space,
+  Typography,
+  Spin,
+  Divider,
+  message,
+} from "antd";
+import Highlighter from "react-highlight-words";
+import { SearchOutlined } from "@ant-design/icons";
+import { useTranslation } from "react-i18next";
+import firebase from "../../firebaseConfig";
+
+const db = firebase.firestore();
+const { Title } = Typography;
+
+const UsedCoupons = ({ isTesting }) => {
+  const { t } = useTranslation();
+  const [usedCoupons, setUsedCoupons] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [searchState, setSearchState] = useState({
+    searchText: "",
+    searchedColumn: "",
+  });
+  const docID = useRef(null);
+  let searchInput;
+
+  const getDocId = async () => {
+    // get the id of the doc to fetch later in snapshot
+    const res = await db
+      .collection("sampleSoldCoupons")
+      .where("compName", "==", "Arby's")
+      .get();
+    const data = res.docs.map((x) => x.id);
+    [docID.current] = data;
+  };
+
+  useEffect(() => {
+    getDocId().then(() => {
+      const unsubscribe = db
+        .collection("sampleSoldCoupons") // this collection is a test, we don't have the real collection yet
+        .doc(docID.current)
+        .collection("coupons")
+        .where("isActive", "==", false)
+        .onSnapshot((snapshot) => {
+          const dataArr = [];
+          snapshot.forEach((doc) => {
+            dataArr.push({
+              ...doc.data(),
+              key: doc.id,
+              // convert the timestamp into a usable date
+              date:
+                doc.data().markDate &&
```

Comment:
the && will cause it to return false or true i think ?

Reply:
Yes without the check, it comes up as undefined so that's required to work with date for timestamps

Reply:
yeah but in that case the date will have a true or false value not the date itself can u check the value of the date ?

Reply:
No, date is there, there is no true value for both of them, it just check if it exist or not. The dates are shown in the table

Reply:
So the result of this check is only date